velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp

```diff
@@ -268,3 +269,23 @@ TEST_F(ParquetReaderTest, parseIntDecimal) {
     EXPECT_EQ(b[index + 1], expectValues[i]);
   }
 }
+
+TEST_F(ParquetReaderTest, intMultipleFilters) {
+  // int BETWEEN 102 AND 120 AND bigint BETWEEN 900 AND 1006
```

Comment:
What about this test failed before ? Why is it a reasonable test for this fix ?

Reply:
Yes, prior this change, the test would fail with the following error: 
```
Expected equality of these values:
  total
    Which is: 10
  expected->size()
    Which is: 5
```
This is because we expected 5 values in return - this filter would only keep 5 records from `int.parquet` file. However, we had to scan 10 records for it so the `part` would return 10 making the subsequent assertion fail. This also causes some odd errors depending on the filter like mismatch in values.